stream_consumer.go

```diff
@@ -31,10 +28,54 @@ type StreamEndpointConfig struct {
 	backoffMaxDelay time.Duration
 }
 
-type Consumer struct {
-	StreamName string
-	EvtChan    chan *stream.Event
+type StreamConsumer interface {
+	StreamName() string
+	EvtChan() chan *stream.Event
+	Stop()
+	streamEndpoint() *StreamEndpoint
+}
+
+type registeredConsumer struct {
+	StreamConsumer
+	g *Gaz
+}
+
+func (c *registeredConsumer) Stop() {
```

Comment:
Who is calling this?
If it's a non public struct and it's not implementing an interface, should it be `stop()` instead?

Reply:
If nobody is calling Stop(), then do we need the registeredConsumer or can we directly use a map[StreamEndpoint]map[*StreamConsumer]struct{}{}?

Reply:
it is implementing the StreamConsumer API, so this is the exposed to the clients.

Reply:
when the client calls Stop()
- if the stream was created the old way (consumer) --> it simply stops the stream
- if the stream was created the new way (registeredConsumer) --> it stops the stream and deregisters it